contracts/modules/delegates/IexecPoco2Delegate.sol

```diff
@@ -1,24 +1,31 @@
 // SPDX-FileCopyrightText: 2020-2024 IEXEC BLOCKCHAIN TECH <contact@iex.ec>
 // SPDX-License-Identifier: Apache-2.0
 
-pragma solidity ^0.6.0;
-pragma experimental ABIEncoderV2;
-
-import "@iexec/solidity/contracts/ERC1154/IERC1154.sol";
-import "./IexecERC20Core.sol";
-import "./SignatureVerifier.sol";
-import "../DelegateBase.sol";
-import "../interfaces/IexecPoco2.sol";
+pragma solidity ^0.8.0;
+
+import {Math} from "@openzeppelin/contracts-v5/utils/math/Math.sol";
+import {IOracleConsumer} from "../../external/interfaces/IOracleConsumer.sol";
+import {IexecLibCore_v5} from "../../libs/IexecLibCore_v5.sol";
+import {IexecLibOrders_v5} from "../../libs/IexecLibOrders_v5.sol";
+import {DelegateBase} from "../DelegateBase.v8.sol";
+import {IexecPoco2} from "../interfaces/IexecPoco2.v8.sol";
+import {IexecEscrow} from "./IexecEscrow.v8.sol";
+import {SignatureVerifier} from "./SignatureVerifier.v8.sol";
+
+contract IexecPoco2Delegate is IexecPoco2, DelegateBase, IexecEscrow, SignatureVerifier {
+    modifier onlyScheduler(bytes32 _taskId) {
+        require(_msgSender() == m_deals[m_tasks[_taskId].dealid].workerpool.owner);
+        _;
+    }
```

Comment:
Can visibilities be set to external or something else blocks it?

Reply:
I prefer doing that separately because I also want to sort functions by visibility.